conan/internal/api/select_pattern.py

```diff
@@ -15,11 +16,21 @@ def split(s, c, default=None):
             return tokens[0], default
 
         recipe, package = split(expression, ":")
+        self.raw = expression
         self.ref, rrev = split(recipe, "#", rrev)
+        ref, user_channel = split(self.ref, "@")
+        self.name, self.version = split(ref, "/")
+        self.user, self.channel = split(user_channel, "/")
         self.rrev, _ = split(rrev, "%")
         self.package_id, prev = split(package, "#", prev)
         self.prev, _ = split(prev, "%")
 
+    @property
+    def is_complete_ref(self):
```

Comment:
I'd say this is a bit misleading, it doesn't check user/channel, which could be patterns, and also name and version could be patterns. I think I'd move the condition directly into the ``ListPattern.mode()``